components/collator/src/comparison.rs

```diff
@@ -91,33 +95,24 @@ impl Collator {
             + DataProvider<CanonicalDecompositionTablesV1Marker>
             + ?Sized,
     {
-        let locale = {
-            // Remove irrelevant extensions, i.e. everything but -u-co-.
-            //
-            // TODO: Revisit at least the default mapping here
-            // once the provider can perform fallback into more
-            // general locale. Once the provider can fall back to
-            // the default, the code here that omits the explicit
-            // variant if it matches the default should become
-            // unnecessary.
-            let original_locale = locale;
-            let mut filtered_locale = alloc::borrow::Cow::Borrowed(locale);
-
-            let key = icu_locid::extensions_unicode_key!("co");
-            if let Some(variant) = original_locale.get_unicode_ext(&key) {
-                let zh = filtered_locale.language() == icu_locid::subtags_language!("zh");
-                let sv = filtered_locale.language() == icu_locid::subtags_language!("sv");
-                // Omit the explicit collation variant if it is the default.
-                // "standard" is the default for all languages except zh and sv.
-                if !((!zh && !sv && variant == value!("standard"))
-                    || (zh && variant == value!("pinyin"))
-                    || (sv && variant == value!("reformed")))
-                {
-                    filtered_locale.to_mut().set_unicode_ext(key, variant);
-                }
+        let mut locale = Cow::Borrowed(locale);
+
+        const KEY: Key = key!("co");
+
+        if let Some(variant) = locale.get_unicode_ext(&KEY) {
+            let zh = locale.language() == language!("zh");
+            let sv = locale.language() == language!("sv");
+            // Remove the explicit collation variant if it is the default.
+            // "standard" is the default for all languages except zh and sv.
```

Comment:
I thought we established that the vertical fallback provider does do this for us.

Can we write a test case if we don't already have one and ensure that it passes, rather than shooting in the dark with a limited understanding?

Also, this is not a 1.2 blocker.